File: src/main/java/com/pedigree/ui/QuickSearchDialog.java
package com.pedigree.ui;

import com.pedigree.model.Individual;
import com.pedigree.search.QuickSearchService;
import com.pedigree.storage.ProjectRepository;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import javafx.geometry.Insets;
import javafx.scene.Scene;
import javafx.scene.control.ListView;
import javafx.scene.control.TextField;
import javafx.scene.layout.BorderPane;
import javafx.stage.Modality;
import javafx.stage.Stage;

import java.util.List;
import java.util.function.Consumer;

public class QuickSearchDialog {
    private final ProjectRepository.ProjectData data;
    private final Consumer<String> onChosen;

    public QuickSearchDialog(ProjectRepository.ProjectData data, Consumer<String> onChosen) {
        this.data = data;
        this.onChosen = onChosen;
    }

    public void show() {
        Stage stage = new Stage();
        stage.initModality(Modality.APPLICATION_MODAL);
        stage.setTitle("Quick Search");

        BorderPane root = new BorderPane();
        root.setPadding(new Insets(10));

        TextField query = new TextField();
        query.setPromptText("Type a name...");
        ListView<Individual> list = new ListView<>();
        ObservableList<Individual> items = FXCollections.observableArrayList();
        list.setItems(items);

        root.setTop(query);
        root.setCenter(list);

        QuickSearchService service = new QuickSearchService(data);
        query.textProperty().addListener((obs, o, n) -> {
            List<Individual> res = service.findByName(n != null ? n : "");
            items.setAll(res);
        });

        list.setOnMouseClicked(e -> {
            Individual sel = list.getSelectionModel().getSelectedItem();
            if (sel != null) {
                if (onChosen != null) onChosen.accept(sel.getId());
                stage.close();
            }
        });

        stage.setScene(new Scene(root, 400, 500));
        stage.showAndWait();
    }
}

File: src/main/kotlin/com/family/tree/ui/QuickSearchDialog.kt
package com.family.tree.ui

import com.family.tree.model.Individual
import com.family.tree.search.QuickSearchService
import com.family.tree.storage.ProjectRepository
import javafx.collections.FXCollections
import javafx.collections.ObservableList
import javafx.geometry.Insets
import javafx.scene.Scene
import javafx.scene.control.ListView
import javafx.scene.control.TextField
import javafx.scene.layout.BorderPane
import javafx.stage.Modality
import javafx.stage.Stage
import java.util.function.Consumer

class QuickSearchDialog(private val data: ProjectRepository.ProjectData, private val onChosen: Consumer<String>?) {

    fun show() {
        val stage = Stage()
        stage.initModality(Modality.APPLICATION_MODAL)
        stage.title = "Quick Search"

        val root = BorderPane()
        root.padding = Insets(10.0)

        val query = TextField()
        query.promptText = "Type a name..."
        val list = ListView<Individual>()
        val items: ObservableList<Individual> = FXCollections.observableArrayList()
        list.items = items

        root.top = query
        root.center = list

        val service = QuickSearchService(data)
        query.textProperty().addListener { _, _, n ->
            val res = service.findByName(n ?: "")
            items.setAll(res)
        }

        list.setOnMouseClicked {
            val sel = list.selectionModel.selectedItem
            if (sel != null) {
                onChosen?.accept(sel.id)
                stage.close()
            }
        }

        stage.scene = Scene(root, 400.0, 500.0)
        stage.showAndWait()
    }
}
